code/modules/atmospherics/machinery/components/unary_devices/machine_connector.dm

```diff
@@ -41,7 +41,7 @@
 	RegisterSignal(connected_machine, COMSIG_MOVABLE_PRE_MOVE, PROC_REF(pre_move_connected_machine))
 	RegisterSignal(connected_machine, COMSIG_MOVABLE_MOVED, PROC_REF(moved_connected_machine))
 	RegisterSignal(connected_machine, COMSIG_MACHINERY_DEFAULT_ROTATE_WRENCH, PROC_REF(wrenched_connected_machine))
-	RegisterSignal(connected_machine, COMSIG_QDELETING, PROC_REF(deconstruct_connected_machine))
+	RegisterSignal(connected_machine, COMSIG_OBJ_DECONSTRUCT, PROC_REF(deconstruct_connected_machine))
```

Comment:
imo, qdeleting should just call its own handler which performs relocate_airs() and such (basically the same event handler but without qdel'ing itself)

deconstruct handler should then unregister qdel before qdeling

Reply:
> qdeleting should just call its own handler which performs relocate_airs()

We want relocation of airs only upon deconstruction and not deletion else that would be a side effect.

But you right. Both deletion and deconstruction share common code except for relocate_airs().

I'll fix that 👍